src/kubernetes.cc

```diff
@@ -115,6 +117,51 @@ KubernetesReader::KubernetesReader(const Configuration& config,
                                    HealthChecker* health_checker)
     : config_(config), environment_(config), health_checker_(health_checker) {}
 
+MetadataStore::Metadata KubernetesReader::GetMetadataOnly(
```

Comment:
The name of this function, and what it does, confuses me slightly. Is the goal to "extract" a metadata struct from a Kubernetes resource? If so, can we either add a comment, or update the name slightly to hint at the extraction / conversion?

Reply:
Just `GetMetadata` sounds good to me.

Reply:
done